scripts/pf2e-extempore-effects.js

```diff
@@ -245,7 +305,74 @@ const onUpdateWorldTime_Wrapper = (wrapped, ...args) => {
+  // let's prompt the user if they want to continue to requested time or stop at the expiry time.
+  if (expireEffectMessageShown && newWorldTime < requestedNewWorldTime) {
+    const dialogTitle = `${localize('.dialog.expired-effects.title')} ${lastExpiryTime}`;
+    let dialogContent = `<span><h6>${localize('.dialog.expired-effects.content.header')}</h6><ul>`;
+
+    for (const effect of effectsExpired) dialogContent += `<li>${effect.name} (${effect.actor.name})</li>`;
+
+    dialogContent += `</ul>${localize('.dialog.expired-effects.content.message-time')} ${formatTime(newWorldTime - oldWorldTime)}.</span><span>${localize('.dialog.expired-effects.content.message-continue')}<br>${formatTime(requestedNewWorldTime - newWorldTime)}?</span>`
+
+    promptUserToContinue(requestedNewWorldTime, newWorldTime, oldWorldTime, dialogTitle, dialogContent);
+  }
+
+  // Play the notification sound if a chat notification was made
+  const effectNotificationSoundSetting = game.settings.get(MODULE_ID, 'notifications-sound-for-expired-effects')
+  if (expireEffectMessageShown && effectNotificationSoundSetting.trim() !== '') {
+    foundry.audio.AudioHelper.play({src: effectNotificationSoundSetting});
+  }
+
+  args[0] = newWorldTime
+  wrapped(...args);
+}
+
+function promptUserToContinue(requestedNewWorldTime, newWorldTime, oldWorldTime, dialogTitle, dialogContent) {
+  new foundry.applications.api.DialogV2({
+    window: {
+      title: dialogTitle,
+      icon: 'fa-regular fa-alarm-exclamation'
+    },
+    content: dialogContent,
+    buttons: [{
+      icon: 'fas fa-check',
+      action: "yes",
+      label: localize('.dialog.expired-effects.buttons.yes'),
+      callback: async () => {
+        await game.time.advance(0) // stupid hack, without this the time doesn't advance properly?
+        await game.time.advance((requestedNewWorldTime - newWorldTime));
+      },
+      default: true
+    }, {
+      icon: 'fas fa-times',
+      action: "no",
+      label: localize('.dialog.expired-effects.buttons.no'),
+    }]
+  }).render({force: true});
+}
+
+
+function formatTime(seconds) {
+  const weeks = Math.floor(seconds / (7 * 24 * 60 * 60));
+  seconds %= 7 * 24 * 60 * 60;
+  const days = Math.floor(seconds / (24 * 60 * 60));
+  seconds %= 24 * 60 * 60;
+  const hours = Math.floor(seconds / (60 * 60));
+  seconds %= 60 * 60;
+  const minutes = Math.floor(seconds / 60);
+  seconds %= 60;
+
+  const parts = [];
+  if (weeks > 0) parts.push(`${weeks} ${localize(weeks > 1 ? '.time.weeks' : '.time.week')}`);
+  if (days > 0) parts.push(`${days} ${localize(days > 1 ? '.time.days' : '.time.day')}`);
+  if (hours > 0) parts.push(`${hours} ${localize(hours > 1 ? '.time.hours' : '.time.hour')}`);
+  if (minutes > 0) parts.push(`${minutes} ${localize(minutes > 1 ? '.time.minutes' : '.time.minute')}`);
+  if (seconds > 0) parts.push(`${seconds} ${localize(seconds > 1 ? '.time.seconds' : '.time.second')}`);
```

Comment:
this will be badly localized because different languages have lots of exceptions to how their times are written down... but I don't see a better solution so let's keep it

Reply:
Ya, I wasn't sure about that either, but I figured it was better to localize vs not localizing it at all.
Granted, I think most of the time you are going to deal with Minutes and Hours (even advancing a day will first break down into hours/minutes.

But I wanted to cover weeks/days since the World Clock does support "weeks".